File: scratch/update_procedures.py
import re

with open('src/constants/procedures.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Add image property to Feature if not present
if 'image?: string;' not in content:
    content = content.replace('effects?: string[];', 'effects?: string[];\n  image?: string;')

categories = [
    ('thermage', '/images/lifting_texture.png'),
    ('skin-booster', '/images/booster_texture.png'),
    ('petit', '/images/petit_texture.png'),
    ('laser', '/images/laser_texture.png'),
]

for cat, img in categories:
    # Match the block for the category
    pattern = re.compile(rf"(slug:\s*'{cat}'.*?)(?=\s+slug:\s*'.*?'|\Z)", re.DOTALL)
    match = pattern.search(content)
    if match:
        block = match.group(1)
        # Replace icon: '...', with icon: '...', image: '...'
        new_block = re.sub(r"(icon:\s*'.*?',)", rf"\1\n        image: '{img}',", block)
        content = content[:match.start()] + new_block + content[match.end():]

with open('src/constants/procedures.ts', 'w', encoding='utf-8') as f:
    f.write(content)

File: scratch/update_procedures.js
const fs = require('fs');
let content = fs.readFileSync('src/constants/procedures.ts', 'utf8');

if (!content.includes('image?: string;')) {
    content = content.replace('effects?: string[];', 'effects?: string[];\n  image?: string;');
}

const categories = [
    { cat: 'thermage', img: '/images/lifting_texture.png' },
    { cat: 'skin-booster', img: '/images/booster_texture.png' },
    { cat: 'petit', img: '/images/petit_texture.png' },
    { cat: 'laser', img: '/images/laser_texture.png' }
];

for (const { cat, img } of categories) {
    const regex = new RegExp(`(slug:\\s*'${cat}'.*?)(?=\\s+slug:\\s*'.*?'|$)`, 's');
    const match = regex.exec(content);
    if (match) {
        let block = match[1];
        block = block.replace(/(icon:\s*'.*?',)/g, `$1\n        image: '${img}',`);
        content = content.substring(0, match.index) + block + content.substring(match.index + match[1].length);
    }
}

fs.writeFileSync('src/constants/procedures.ts', content, 'utf8');
console.log('Done!');
